core/collector.class.inc.php

```diff
@@ -716,35 +716,51 @@ public function Synchronize($iMaxChunkSize = 0)
 		$sResult = self::CallItopViaHttp("/synchro/synchro_exec.php?login_mode=$sLoginform",
 			$aData);
 
-		$iErrorsCount = 0;
+		$iErrorsCount = $this->ParseSynchroExecOutput($sResult);
+
+		return ($iErrorsCount == 0);
+	}
+
+	public function ParseSynchroExecOutput($sResult) : int
+	{
 		if (preg_match_all('|<input type="hidden" name="loginop" value="login"|', $sResult, $aMatches)) {
 			// Hmm, it seems that the HTML output contains the login form !!
 			Utils::Log(LOG_ERR, "Failed to login to iTop. Invalid (or insufficent) credentials.");
 			$this->sErrorMessage .= "Failed to login to iTop. Invalid (or insufficent) credentials.\n";
-			$iErrorsCount = 1;
-		} else {
-			if (preg_match_all('/Objects (.*) errors: ([0-9]+)/', $sResult, $aMatches)) {
-				foreach ($aMatches[2] as $idx => $sErrCount) {
-					$iErrorsCount += (int)$sErrCount;
-					if ((int)$sErrCount > 0) {
-						Utils::Log(LOG_ERR, "Synchronization of data source '{$this->sSourceName}' answered: {$aMatches[0][$idx]}");
-						$this->sErrorMessage .= $aMatches[0][$idx]."\n";
-					}
+			return 1;
+		}
+
+		$iErrorsCount = 0;
+		if (preg_match_all('/Objects (.*) errors: ([0-9]+)/', $sResult, $aMatches)) {
+			foreach ($aMatches[2] as $idx => $sErrCount) {
+				$iErrorsCount += (int)$sErrCount;
+				if ((int)$sErrCount > 0) {
+					Utils::Log(LOG_ERR, "Synchronization of data source '{$this->sSourceName}' answered: {$aMatches[0][$idx]}");
+					$this->sErrorMessage .= $aMatches[0][$idx]."\n";
 				}
-			} else {
-				Utils::Log(LOG_ERR, "Synchronization of data source '{$this->sSourceName}' failed.");
-				$this->sErrorMessage .= $sResult;
-				$iErrorsCount = 1;
 			}
+		} else {
+			Utils::Log(LOG_ERR, "Synchronization of data source '{$this->sSourceName}' failed.");
+			$this->sErrorMessage .= $sResult;
+			return 1;
 		}
+
+		if (($iErrorsCount === 0) && preg_match_all('/<p>ERROR: (.*)\./', $sResult, $aMatches)) {
+			foreach ($aMatches[1] as $idx) {
+				Utils::Log(LOG_ERR, "Synchronization of data source '{$this->sSourceName}' answered: $idx");
+				$this->sErrorMessage .= $idx."\n";
+				return 1;
+			}
+		}
+
 		if ($iErrorsCount == 0) {
 			Utils::Log(LOG_INFO, "Synchronization of data source '{$this->sSourceName}' succeeded.");
 		}
 
-		return ($iErrorsCount == 0);
+		return $iErrorsCount;
 	}
 
-	public static function ParseSynchroOutput($sTrimmedOutput, $bDetailedOutput) : string
+	public static function ParseSynchroImportOutput($sTrimmedOutput, $bDetailedOutput) : string
```

Comment:
```suggestion
	/**
	 * @since 1.3.1 Method renamed from `ParseSynchroOutput` to `ParseSynchroImportOutput`
	 */
	public static function ParseSynchroImportOutput($sTrimmedOutput, $bDetailedOutput) : string
```

I was afraid the renaming could impact any usage of this method in other collectors, but I didn't find any. So it's alright.
Just warn that it was renamed.

Reply:
you are right. no need to rename. i will revert to avoid any regression for those that write their own collectors...

Reply:
See my suggestion from another thread:

```php
	/**
	 * @deprecated 1.3.1 N°6771
	 * @see static::ParseSynchroImportOutput
	 */
	public static function ParseSynchroOutput($sTrimmedOutput, $bDetailedOutput) : string {
		// log a deprecation message
		// Utils::Log(...);
		return static::ParseSynchroImportOutput($sTrimmedOutput, $bDetailedOutput);
	}
	
	/**
	 * PHPDOC ..
	 */
	public static function ParseSynchroImportOutput($sTrimmedOutput, $bDetailedOutput) : string {
```